javascript/src/main/components/OfficeHours/OfficeHourForm.js

```diff
@@ -20,13 +20,18 @@ const OfficeHourForm = ({ createOfficeHour, /*updateOfficeHour, /*existingOffice
         notes: ""
     }
 
-    const [officeHour, setOfficeHour] = useState(emptyOfficeHour);
+    const [officeHour, setOfficeHour] = useState(existingOfficeHour||emptyOfficeHour);
 
     const handleOnSubmit = (e) => {
         e.preventDefault();
         var isValid = checkInputs(); 
         if (isValid) {
-            createOfficeHour(officeHour);
+            if (createOfficeHour) {
```

Comment:
Is this how edit has been implemented for the other pages?

Reply:
Edit has been implemented similarly compared to other pages, but that does mean that it has the same implementation.

Reply:
This looks ok to me.    The `if (createOfficeHour)` is used to see whether the component that invoked this passed a value for `createOfficeHour` or not.  If they didn't, then it's assumed that they passed a value for `editOfficeHour`.   That's how the component knows whether it's being used for creating or editing.